src/components/UI/MainInput.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <div class="wrapper">
+    <label :for="iputCat" class="input-label"><slot /></label>
+    <input
+      class="input-box"
+      required
+      :type="iputCat"
+      placeholder="..."
+      :value="value"
+      :name="iputCat"
+      @input="$emit('input', $event.target.value)"
+      @blur="$emit('blurTouched')"
+    />
+  </div>
+</template>
+
+<script>
+export default {
+  name: "MainInput",
+  props: {
+    value: {
+      type: String,
+      default: "",
+    },
+    inputType: {
+      type: String,
+      default: "text",
+    },
+  },
+  computed: {
+    iputCat() {
+      switch (this.inputType) {
+        case "text":
+          return "text";
+        case "password":
+          return "password";
+        case "date":
+          return "date";
+        case "email":
+          return "email";
+        case "month":
+          return "month";
+        case "number":
+          return "number";
+        default:
+          return "text";
+      }
```

Comment:
We might as well just bind the prop directly to the input type, considering this computed return the prop's value 😃

Reply:
now that i'm looking at this in the daylight it makes 110% sense.  🗡️  thank you!!!